tests/ui/arithmetic_side_effects.rs

```diff
@@ -521,6 +521,10 @@ pub fn issue_11393() {
     example_rem(x, maybe_zero);
 }
 
+pub fn issue_11266<const N: usize>() {
+    let _ = N + 1;
+}
```

Comment:
Why don't we want to lint this? It doesn't seem like rustc generates a warning for [the example code](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=98e39f93d0509c0b4e9f41f9211e8cf7), or am I missing something?

Something like `N + random()` could still run into the same issues in that it could overflow without the compiler possibly knowing it, can't it?

Reply:
Changed to only ignore when literals are involved.

Reply:
I see.  But even then, rustc doesn't seem to catch the case with literals in my example and doesn't issue a warning for it.

Maybe I'm misunderstanding why this change is done, but the way I understood it was, "this lint doesn't need to warn const generic operations because rustc has its own `arithmetic_overflow` lint that is more accurate for consts and will issue a warning/error if it sees overflow after monomorphization when all const generic parameters are instantiated with concrete values", but it doesn't seem to do that, so now there is no warning for `N + 1` emitted from either side even for `usize::MAX`.

Reply:
> I see. But even then, rustc doesn't seem to catch the case with literals in my example and doesn't issue a warning for it.

![Capture d’écran de 2024-04-24 11-39-06](https://github.com/rust-lang/rust-clippy/assets/17877264/2a8d3b9c-2098-43e4-89bb-a43d6df47b15)

![Capture d’écran de 2024-04-24 11-39-26](https://github.com/rust-lang/rust-clippy/assets/17877264/307e23af-6432-46aa-930b-745557220178)

Reply:
The top one is only a runtime panic though. And the bottom one will be gone with this change presumably? So that would leave us with no compile time warnings/errors

Reply:
Yes, you are right. Thanks for pointing this out